pkg/apis/networking/validation/validation.go

```diff
@@ -381,22 +385,34 @@ func validateHTTPIngressPath(path *networking.HTTPIngressPath, fldPath *field.Pa
 		}
 	}
 
-	allErrs = append(allErrs, validateIngressBackend(&path.Backend, fldPath.Child("backend"))...)
+	allErrs = append(allErrs, validateIngressBackend(&path.Backend, fldPath.Child("backend"), opts)...)
 	return allErrs
 }
 
 // validateIngressBackend tests if a given backend is valid.
-func validateIngressBackend(backend *networking.IngressBackend, fldPath *field.Path) field.ErrorList {
+func validateIngressBackend(backend *networking.IngressBackend, fldPath *field.Path, opts IngressValidationOptions) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	// All backends must reference a single local service by name, and a single service port by name or number.
-	if len(backend.ServiceName) == 0 {
-		return append(allErrs, field.Required(fldPath.Child("serviceName"), ""))
-	}
-	for _, msg := range apivalidation.ValidateServiceName(backend.ServiceName, false) {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceName"), backend.ServiceName, msg))
+	hasResourceBackend := backend.Resource != nil
+	hasServiceBackend := len(backend.ServiceName) > 0 || backend.ServicePort.IntVal != 0 || len(backend.ServicePort.StrVal) > 0
+
+	switch {
+	case hasResourceBackend && hasServiceBackend:
+		return append(allErrs, field.Invalid(fldPath, "", "cannot set both resource and service backends"))
+	case hasResourceBackend && !opts.allowResourceBackend:
+		return append(allErrs, field.Forbidden(fldPath.Child("resource"), "cannot be set on this object"))
```

Comment:
thinking about this more, I wonder if there's a clearer error message, like "not supported; only service backends are supported in this version"